### PR TITLE
CSS-3375: replace ingress relation with nginx-route

### PR DESCRIPTION
**Pull request by kelkawi-a**
This is to address the [deprecation](https://github.com/canonical/nginx-ingress-integrator-operator/pull/54) of the `ingress` interface in the `nginx-ingress-integrator` charm.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -107,9 +109,6 @@ def _on_config_changed(self, event):
             event: The event triggered when the relation changed.
         """
         self.unit.status = WaitingStatus("configuring temporal")
-        self.ingress.update_config(
-            {"service-hostname": self.external_hostname, "tls-secret-name": self.config["tls-secret-name"]}
-        )
```
Comment (frankban) [inline src/charm.py:112]: Do you need to call _require_nginx_route here as well?

File: tests/integration/test_charm.py [hunk 2]
```diff
@@ -6,12 +6,15 @@
...
 import logging
+import socket
+import unittest.mock
 from pathlib import Path
 
 import pytest
 import pytest_asyncio
 import requests
 import yaml
+from helpers import gen_patch_getaddrinfo
```
Comment (frankban) [inline tests/integration/test_charm.py:17]: This should to the third import block for internals.

**Review by frankban — APPROVED** [review]
Looks good!

File: src/charm.py (hunk 1 above)
Comment (kelkawi-a) [reply src/charm.py:112]: No need, this is reflected in the `test_ingress` integration test.

File: tests/integration/test_charm.py (hunk 2 above)
Comment (kelkawi-a) [reply tests/integration/test_charm.py:17]: The [`isort`](https://pycqa.github.io/isort/) formatter automatically sorts imports by alphabetical order.

**Review by TeodorPt — APPROVED** [review]

File: src/charm.py (hunk 1 above)
Comment (kian99) [reply src/charm.py:112]: Interesting, I wonder how the nginx-route relation knows to refresh its config when the related application's config changes.

**Review by kian99 — APPROVED** [review]

File: src/charm.py (hunk 1 above)
Comment (kelkawi-a) [reply src/charm.py:112]: I believe it relies on a `relation_changed` hook:

`
self._charm.framework.observe(self._charm.on[self._nginx_route_relation_name].relation_changed, self._config_reconciliation)
`

**Pull request merged**